### PR TITLE
Search style

### PR DESCRIPTION
**Pull request by danthewolfe**
* added search box and button (that work).
* toggle button colors on grid/search buttons
* style the modals for search
* add search box to collection homepages

### CODE REVIEW

**Comment by jgondron** [comment]
:+1:

File: app/assets/javascripts/components/pages/ItemsSearchPage.jsx [hunk 1]
```diff
@@ -31,7 +31,10 @@ var ItemsSearchPage = React.createClass({
     } else {
       this.loadRemoteCollection(this.props.collection);
     }
-    this.loadSearchResults(this.props.hits + "?q=" + this.props.query_parameters);
+    this.loadSearchResults(
+      this.props.hits
+      + "?q=" + this.props.searchTerm
```
Comment (jaronkk) [inline app/assets/javascripts/components/pages/ItemsSearchPage.jsx]: This should use encodeURIComponent() around the search term

File: app/assets/javascripts/components/search/SearchBox.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  onChange: function(e) {
+    this.setState({searchTerm: this.refs.searchBox.getValue()});
+  },
+
+  onClick: function(e) {
+    window.location.assign(this.searchUrl());
+  },
+
+  searchUrl: function() {
+    var url = window.location.origin + "/" +this.props.collection.id + "/" + this.props.collection.slug + "/search?q=" + this.state.searchTerm
```
Comment (jaronkk) [inline app/assets/javascripts/components/search/SearchBox.jsx]: This should use encodeURIComponent() around the search term

**Comment by jaronkk** [comment]
Is the SearchSort component being used in any way right now, or is it just a placeholder?  Both SearchSort and SearchBox modify the window.location, so it seems like there should be a single parent component responsible for handling the URL by responding to events from the child components.

**Comment by danthewolfe** [comment]
SearchSort is not being used at the moment. Right now it is placeholder. I wanted to get the other changes merged in first.

**Comment by jaronkk** [comment]
:+1:

**Pull request merged**